improver_tests/lightning/test_LightningMultivariateProbability_USAF2024.py

```diff
@@ -0,0 +1,297 @@
+    returns the error message this will trigger"""
+    li_cube.rename("LI")
+    return r"No cube named temperature_difference_between_ambient_air_and_air_lifted_adiabatic.*"
+
+
+def break_pw_name(cape_cube, precip_cube, cin_cube, li_cube, pw_cube):
+    """Modifies pw_cube name to be changed to appear missing and
+    returns the error message this will trigger"""
+    pw_cube.rename("PW")
+    return r"No cube named precipitable_water found in .*"
+
+
+def break_cape_time(cape_cube, precip_cube, cin_cube, li_cube, pw_cube):
+    """Modifies cape_cube time points to be incremented by 1 second and
+    returns the error message this will trigger"""
+    cape_cube.coord("time").points = cape_cube.coord("time").points + 1
+    return r"The .*energy time .* should be valid at the precipitation_accumulation cube lower .*"
+
+
+def break_precip_window(cape_cube, precip_cube, cin_cube, li_cube, pw_cube):
+    """Modifies upper bound on precip_cube time coord to be incremented by 1 second and
+    returns the error message this will trigger"""
+    bounds = precip_cube.coord("time").bounds
+    precip_cube.coord("time").bounds = (bounds[0][0], bounds[0][1] + 1)
+    return r"Precipitation_accumulation cube time window must be three hours, not .*"
+
+
+def break_reference_time(cape_cube, precip_cube, cin_cube, li_cube, pw_cube):
+    """Modifies cape_cube time points to be incremented by 1 second and
+    returns the error message this will trigger"""
+    precip_cube.coord("forecast_reference_time").points = (
+        precip_cube.coord("forecast_reference_time").points + 1
+    )
+    return r".* and .* do not have the same forecast reference time"
+
+
+def break_coordinates(cape_cube, precip_cube, cin_cube, li_cube, pw_cube):
+    """Modifies the first latitude point on the precip_cube (adds one degree)
+    and returns the error message this will trigger"""
+    points = list(precip_cube.coord("latitude").points)
+    points[0] = points[0] + 1
+    precip_cube.coord("latitude").points = points
+    return ".* and .* do not have the same spatial coordinates"
+
+
+@pytest.mark.parametrize(
+    "breaking_function",
+    (
+        break_cape_name,
+        break_precip_name,
+        break_cin_name,
+        break_li_name,
+        break_pw_name,
+        break_cape_time,
+        break_precip_window,
+        break_reference_time,
+        break_coordinates,
+    ),
+)
+def test_exceptions(
+    cape_cube, apcp_cube, cin_cube, liftidx_cube, pwat_cube, breaking_function
+):
+    """Tests that a suitable exception is raised when the  cube meta-data does
+    not match what is expected"""
+    error_msg = breaking_function(
+        cape_cube, apcp_cube, cin_cube, liftidx_cube, pwat_cube
+    )
+    with pytest.raises(ValueError, match=error_msg):
+        LightningMultivariateProbability_USAF2024()(
+            CubeList([cape_cube, apcp_cube, cin_cube, liftidx_cube, pwat_cube])
+        )
```

Comment:
If you adopt my suggestion to check the input units, you can also test breaking those here as well.

Reply:
Added a check into the unit tests to ensure that the units are checked for consistency.